python/paddle/fluid/compiler.py

```diff
@@ -193,17 +217,15 @@ def with_data_parallel(self,
         self._exec_strategy = exec_strategy
         self._loss_name = loss_name
         self._share_vars_from = share_vars_from
-        if self._exec_strategy is None:
-            self._exec_strategy = ExecutionStrategy()
-        if self._build_strategy is None:
-            self._build_strategy = BuildStrategy()
-        if places is not None:
-            if not isinstance(places, (list, tuple)):
-                places = [places]
-            self._places = places
-        else:
-            self._places = None
-        self._build_strategy.is_distribution = _is_pserver_mode(self._program)
+        self._places = places
+
+        if _has_backward_op(self._graph):
+            assert self._loss_name is not None, "The loss_name should be set here."
```

Comment:
For training, the default loss@GRAD should be `1/dev_count`, this is set in ParallelExecutor, and this process needs to use `loss_name `, otherwise, the loss@GRAD is 1.